src/comp/pages/UserDetailsPage.tsx

```diff
@@ -0,0 +1,171 @@
+  updateFollowingList: (followingIndex: number, userId: number) => void;
+};
+
+const accountsIcons: accountTypeOfElement = {
+  twitter: <Twitter fontSize="small" />,
+  github: <GitHub fontSize="small" />,
+  mailAddress: <Mail fontSize="small" />,
+};
+
+const url: accountType = {
+  twitter: "https://twitter.com/",
+  github: "https://github.com/",
+  mailAddress: "mailto:",
+};
+
+class UserDetailsPage extends React.Component<
+  userDetailsProps,
+  userDetailsState
+> {
+  constructor(props: userDetailsProps) {
+    super(props);
+    this.state = {
+      userId: 103,
+      name: "unios103",
+      birthDay: 20200415,
+      bio: "unios103だよ！Reactべんつよします。",
+      favoriteLanguage: "TypeScript",
+      profileImageUrl: "https://github.com/unios103.png",
+      accounts: {
+        github: "unios103",
+        twitter: "unios103",
+        mailAddress: "unios103@gmail.com",
+      },
+      showList: ["github", "twitter", "mailAddress"],
+      developmentExperienceId: [7438921],
+      followingIdList: [11, 33],
+      followerIdList: [112],
+      isPhone: window.innerWidth <= 600,
+    };
+  }
+
+  returnAccount = () => {
+    const accounts: accountType = this.state.accounts;
+    const accountElement = this.state.showList.map(
+      (
+        accountName: keyof accountType | keyof accountTypeOfElement,
+        i: number
+      ) => {
+        return (
+          <Link
+            className={styles.iconAndText}
+            color="secondary"
+            key={i}
+            href={url[accountName] + accounts[accountName]}
+          >
+            <p>{accountsIcons[accountName]}</p>
+            {this.state.isPhone ? <></> : <p>{accounts[accountName]}</p>}
+          </Link>
+        );
+      }
+    );
+    return accountElement;
+  };
+
+  handleResize = () => {
+    this.setState({ isPhone: window.innerWidth <= 600 });
+  };
+
+  userDetails = () => {
+    return (
+      <UserDetailsComponent
+        birthDay={this.state.birthDay}
+        favoriteLanguage={this.state.favoriteLanguage}
+        profileImageUrl={this.state.profileImageUrl}
+        followingIdList={this.state.followingIdList}
+        followerIdList={this.state.followerIdList}
+      />
+    );
+  };
+
+  render() {
+    return (
+      <>
+        <EventListener target="window" onResize={this.handleResize} />
+        <div className={styles.profilePageWrapper}>
+          <div className={styles.profileWrapper}>
+            <div className={styles.profileImageWrapper}>
+              <img
+                src={this.state.profileImageUrl}
+                alt={this.state.name}
+                className={styles.profileImage}
+              ></img>
+            </div>
+            <div className={styles.followButton}>
+              <FollowButton
+                userId={this.state.userId}
+                loginUserFollowingIdList={this.props.loginUserFollowingIdList}
+                updateFollowingList={this.props.updateFollowingList}
```

Comment:
`props` の関数を直接渡さずに、一度関数を作り直して結合度を弱めてください
```suggestion
                updateFollowingList={(index, id) => this.props.updateFollowingList(index, id)}
```